src/microbe_stage/PatchRegion.cs

```diff
@@ -0,0 +1,304 @@
+        {
+            var adjacent = Adjacent.First();
+            LinkPatches(Patches[0], adjacent.Patches[adjacent.Patches.Count - 1]);
+        }
+
+        // Patches position configuration
+        for (int i = 0; i < Patches.Count; i++)
+        {
+            if (RegionType is "sea" or "ocean")
+            {
+                Patches[i].ScreenCoordinates = new Vector2(ScreenCoordinates.x + regionMargin,
+                    ScreenCoordinates.y + i * (64f + PatchMargin) + PatchMargin + RegionLineWidth);
+
+                // Random depth for water regions
+                if (i == Patches.Count - 2)
+                {
+                    var depth = Patches[i].Depth;
+                    var seafloor = Patches[i + 1];
+                    Patches[i].Depth[1] = random.Next(depth[0] + 1, depth[1] - 10);
+
+                    seafloor.Depth[0] = Patches[i].Depth[1];
+                    seafloor.Depth[1] = Patches[i].Depth[1] + 10;
+                }
+            }
+
+            if (RegionType == "continent")
+            {
+                if (i % 2 == 0)
+                {
+                    if (i == 0)
+                    {
+                        Patches[i].ScreenCoordinates = new Vector2(ScreenCoordinates.x + regionMargin,
+                            ScreenCoordinates.y + regionMargin);
+                    }
+                    else
+                    {
+                        Patches[i].ScreenCoordinates = new Vector2(ScreenCoordinates.x + regionMargin,
+                            ScreenCoordinates.y + 64f + 2 * PatchMargin);
+                    }
+                }
+                else
+                {
+                    if (i == 1)
+                    {
+                        Patches[i].ScreenCoordinates =
+                            new Vector2(ScreenCoordinates.x + 2 * PatchMargin + 64f + RegionLineWidth,
+                                ScreenCoordinates.y + regionMargin);
+                    }
+                    else
+                    {
+                        Patches[i].ScreenCoordinates =
+                            new Vector2(ScreenCoordinates.x + 2 * PatchMargin + 64f + RegionLineWidth,
+                                ScreenCoordinates.y + 2 * PatchMargin + 64f + RegionLineWidth);
+                    }
+                }
+            }
+
+            if (RegionType is "vents" or "underwater_cave")
+            {
+                Patches[0].ScreenCoordinates = new Vector2(ScreenCoordinates.x + regionMargin,
+                    ScreenCoordinates.y + regionMargin);
+
+                // Caves or vents are the same depth as the adjacent patch
+                Patches[0].Depth[0] = Patches[0].Adjacent.First().Depth[0];
+                Patches[0].Depth[1] = Patches[0].Adjacent.First().Depth[1];
+            }
+        }
+    }
+
+    public void BuildRegion()
+    {
+        // Region size configuration
+        Width += 64f + 2 * PatchMargin + RegionLineWidth;
```

Comment:
I'll also comment on this once, I do not want to see any floats with just a "f" suffix. They always need to have a decimal point like "64.0f".
Once I get around to it I will write a code check that will not let any number followed by f through.

Reply:
why?

Reply:
Because I like uniform code style. I have added the code check already so next time master is merged into this branch, it should start complaining about these things.

Reply:
:|
I dont know whats so uniform about adding a zero when its not needed.

Reply:
In C++ I would write `1.f` but that's not valid syntax. So I write `1.0f` instead. And as I've written / reviewed most of the game's code, that convention is followed (now that I added that check and fixed a few spots I missed) everywhere.